project_weather_app/data_service.py

```diff
@@ -0,0 +1,77 @@
+import requests
+import read_config as rc
+from abc import abstractmethod
+
+
+# interface for DataService
+class IDataService:
+    @abstractmethod
+    def __init__(self, data_source):
+        pass
+
+    @abstractmethod
+    def get_data(self):
+        pass
+    
+    @abstractmethod
+    def print_data(self):
+        pass
+    
+    @abstractmethod
+    def print_status(self):
+        pass
+
+
+class DataServiceFromAPI(IDataService):
+    def __init__(self, data_source):
+        self._url, self._payload = rc.get_config(data_source)
+
+    def get_data(self):
+        try:
+            r = requests.get(self._url, params=self._payload)
+        except Exception:
+            print("Error: Cannot get data with API.")
+            time = []
+            temp = []
+        else:
+            data = r.json().get("hourly", {})
+            time = data["time"]
+            temp = data[self._payload["hourly"]]
+        self.status_code = r.status_code
+        self.time_list = time
+        self.temp_list = temp
+    
+    def print_data(self):
+        for time, temp in zip(self.time_list, self.temp_list):
+            print(time, "", temp)
+    
+    def print_status(self):
+        print("API status code: ", self.status_code)
+
+
+# placeholder for dummy data service from file.
+class DataServiceFromFile(IDataService):
+    def __init__(self, data_source):
+        super().__init__(data_source)
+    def get_data(self):
+        pass
+    def print_data(self):
+        pass
+    def print_status(self):
+        pass
```

Comment:
If you have a file-based mocked service, you should implement it in a way that it reads mocked data from a file. I think currently, this service wouldn't be able to execute anything if i read the code correctly?